app/uk/gov/hmrc/tai/service/PersonService.scala

```diff
@@ -29,9 +28,11 @@ import scala.concurrent.Future
 class PersonService @Inject()(personConnector: PersonConnector) {
 
   def personDetails(nino: Nino)(implicit hc: HeaderCarrier): Future[Person] =
-    personConnector.person(nino) map {
-      case TaiSuccessResponseWithPayload(person: Person) => person
-      case _ =>
-        throw new RuntimeException(s"Failed to retrieve person details for nino ${nino.nino}. Unable to proceed.")
-    }
+    personConnector
+      .person(nino)
+      .recoverWith {
+        case _ =>
+          Future.failed(
```

Comment:
is there any point in this whole method? Could we either move the exception to the PersonConnector.person?